terraso_backend/apps/graphql/schema/story_maps.py

```diff
@@ -231,21 +258,194 @@ def validate(context):
             raise GraphQLNotAllowedException(
                 model_name=Membership.__name__, operation=MutationTypes.UPDATE
             )
+        except IntegrityError as exc:
+            logger.info(
+                "Attempt to mutate an model, but it's not unique",
+                extra={"model": Membership.__name__, "integrity_error": exc},
+            )
+
+            validation_error = ValidationError(
+                message={
+                    NON_FIELD_ERRORS: ValidationError(
+                        message=f"This {Membership.__name__} already exists",
+                        code="unique",
+                    )
+                },
+            )
+            raise GraphQLValidationException.from_validation_error(
+                validation_error, model_name=Membership.__name__
+            )
         except Exception as error:
             logger.error(
                 "Attempt to update Story Map Memberships, but there was an error",
                 extra={"error": str(error)},
             )
             raise GraphQLNotFoundException(model_name=Membership.__name__)
 
-        approved_memberships = [
-            membership["membership"] for membership in memberships if membership["was_approved"]
+        pending_memberships = [
+            membership["membership"] for membership in memberships if not membership["was_approved"]
         ]
-        send_memberships_invite_email(approved_memberships, story_map)
+        send_memberships_invite_email(user, pending_memberships, story_map)
 
         return cls(memberships=[membership["membership"] for membership in memberships])
 
 
+class StoryMapMembershipApproveTokenMutation(BaseUnauthenticatedMutation):
```

Comment:
question: want to make sure I understand this right, is this how this works?

1) memberships can be made for anonymous users and sent via email
2) because anonymous users are not logged in, they can be given a membership token which links their session to a membership
3) every time the token expires, the membership becomes invalid

Is that right? If that's the case, is there any plan to delete anonymous memberships whose associated token has expired?